sdk/textanalytics/ai-text-analytics/sample.env

```diff
@@ -14,4 +14,7 @@ AZURE_CLIENT_SECRET=<client secret used to authenticate to Azure AD>
 
 # Our tests assume that TEST_MODE is "playback" by default. You can
 # change it to "record" to generate new recordings, or "live" to bypass the recorder entirely.
-# TEST_MODE=playback
\ No newline at end of file
+# TEST_MODE=playback
+# Control which authentication method to use. Default is AAD. To use an API key,
+# set it to "APIKey".
+# TEST_AUTH_MODE=AAD
```

Comment:
Interesting!
For other libraries, take service-bus for example, we have all the tests running only with the connection string and a single test specific for AAD to make sure the auth works. Same with storage libraries I believe.

With this new mode, 
- How would you want to manage the recordings? (Would be different for "APIKey" vs "AAD" but the test title remains the same)
- How often do you want to run these tests with different modes? (Live tests would pick the auth mode at random?)

Reply:
@HarshaNalluru the issue is that sometimes a preview/canary deployment fails only some endpoints or methods with AAD. In FR I've just decided to run all tests on both AAD and using API Keys, so I'll see if anything fails and where. Sometimes I want to disable AAD tests for one reason or another, and so it would be nice to have "AAD + API Key" be the default, but optionally be able to just run with API Key".

Another question is about services that support more than just these two authentication methods, for example anonymous auth in storage.

Reply:
@HarshaNalluru 
- we can generate tests [dynamically](https://mochajs.org/#dynamically-generating-tests) if we need to. However, I am not able to see why we would ever need to record tests in both modes.
- this can be a discussion for later with the team. For text analytics, I often need to run the tests in both modes closer to our releases.

@willmtemple 
> Another question is about services that support more than just these two authentication methods, for example anonymous auth in storage.

I would imagine this approach is simple enough to accommodate those scenarios as well. Each package needs to provide an adapter that creates the credential the user requested.

Reply:
> However, I am not able to see why we would ever need to record tests in both modes.

@deyaaeldeen  Do the recordings stay the same irrespective of the auth mode?